community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/References.java

```diff
@@ -134,4 +140,79 @@ static boolean hasDirectFlag( long groupReference )
         assert groupReference != NO_ID;
         return (groupReference & DIRECT_FLAG) != 0L;
     }
+
+    /**
+     * Encode a node id as a property reference.
+     *
+     * This allows us to encode a node reference as a property reference so that we can do lookups of changes in the
+     * transaction state.
+     * @param reference the reference to encode
+     * @return The encoded reference
+     */
+    static long setNodeFlag( long reference )
```

Comment:
In this case it's sort of confusing, because we in essence take a nodeReference and return a propertyReference. This clashes a bit with the flag concept. Suggestions are welcome!

Reply:
Well, the reference coming in is still gona be a reference to a node, etc. Then we encode some more information into that for later use by us and when we read the reference we kind of say "yeah I know this is a node reference but I'm being promised it's safe for me to use it as a property reference." So we are not turning it into something that it isn't just make promises about additional possibilities.
Is that different perspective
a) Accurate at all?
b) Helpful?

Reply:
updated names of parameters